File: src/Spider/Models/SpiderState.swift
import Foundation

public struct SpiderState: Codable, Equatable {
    public var stock: Pile
    public var foundations: [Pile]
    public var tableau: [Pile]
    public var score: Int
    public var movesCount: Int
    public var timerSeconds: Int
    public var isTimerActive: Bool
    public var hasWon: Bool
    
    public init(
        stock: Pile,
        foundations: [Pile],
        tableau: [Pile],
        score: Int,
        movesCount: Int,
        timerSeconds: Int,
        isTimerActive: Bool,
        hasWon: Bool
    ) {
        self.stock = stock
        self.foundations = foundations
        self.tableau = tableau
        self.score = score
        self.movesCount = movesCount
        self.timerSeconds = timerSeconds
        self.isTimerActive = isTimerActive
        self.hasWon = hasWon
    }
}

File: android/app/src/main/java/com/leah/honeycomb/spider/SpiderState.kt
package com.leah.honeycomb.spider

import com.leah.honeycomb.Pile
import kotlinx.serialization.Serializable

@Serializable
data class SpiderState(
    val stock: Pile = Pile(id = "stock", type = com.leah.honeycomb.PileType.Stock),
    val tableau: List<Pile> = emptyList(),
    val foundations: List<Pile> = emptyList(), // Typically 8 foundations
    val score: Int = 500,
    val movesCount: Int = 0,
    val hasWon: Boolean = false,
    val timerSeconds: Int = 0,
    val isTimerActive: Boolean = false,
    val hintAvailable: Boolean = false
)
